### ISSUE TITLE
Getting started with DevDeck

### ISSUE THREAD

**Issue by pklaus** [jamesridgway/devdeck#1]
Hi jamesridgway, nice project! I like your YouTube video introducing the software. Well done.

When trying out the tool, I came across a few glitches that I wanted to share for future improvement.
The following is based on my experience with DevDeck as of [1.0.1](https://github.com/jamesridgway/devdeck/releases/tag/1.0.1):

1. When running `devdeck` for the first time, I got:

```
FileNotFoundError: [Errno 2] No such file or directory: '/home/pklaus/.devdeck/settings.yml'
```

After creating `~/.devdeck/`, the second attempt did start the tool, creating the default config.

2. When trying to add the `devdeck.decks.volume_deck.VolumeDeck` control, I ran into the issue:

```
  File "/home/pklaus/.pyvenv/py39/lib/python3.9/site-packages/devdeck/decks/volume_deck.py", line 26, in <listcomp>
    selected_output = [output for output in sinks if output.description == self.settings['output']]
KeyError: 'output'
```

So I found out that I need to add a settings section (with the output name to be adjust by the control) like this:

```yaml
[...]
    - key: 1
      name: devdeck.decks.volume_deck.VolumeDeck
      settings:
        output: "WH-1000XM3"
```
It would be nice if there was some documentation on the config format. At least an example yaml file.

3. When running the adjusted configuration, I get:

```
FileNotFoundError: [Errno 2] No such file or directory: '/home/pklaus/.pyvenv/py39/lib/python3.9/site-packages/devdeck/controls/../../assets/font-awesome/volume-up-solid.png'
```

Seems like the icon file (or rather the entire assets folder) is missing in the PyPI distribution of DevDeck.

----

Final words:

I really like this software for:
* its approach to define the style via a lighweight config file,
* use a proper plugin system,
* its ability to provide the framework to use [python-elgato-streamdeck [Github]](https://github.com/abcminiuser/python-elgato-streamdeck) aka. [streamdeck [PyPI]](https://pypi.org/project/streamdeck/) in a much easier way.

**Comment by pklaus** [comment]
A few more observations:

* The README incorrectly states `devdeck.controls.clock_control.CommandControl` instead of `devdeck.controls.clock_control.ClockControl`.

* When setting a custom icon for the `devdeck.controls.command_control.CommandControl`, the image is pasted into a 512x512 px image defined by the renderer. The size of my source image is 72x72 pixels - small, but exactly the pixel dimensions a key of the StreamDeck has. After pasting it in the top left corner of the renderer's image, it looks rather tiny.

**Comment by jamesridgway** [comment]
Hi @pklaus 

Thanks for the feedback!

I just wanted to do a quick reply to let you know that I've seen this and plan to take a look at it later this week (time permitting).

I definitely want to issue an update to resolve the issues you've had.

I thought I'd sorted the image asset issue with the packaging of the image files, but something must still need tweaking on that.

As for the image size point (72px vs 512px). I wonder if there is some scaling/dpi-scaling going on. I know what you mean in that the stream deck docs say its a 72px key size, so a 72px image should fit, but when I was originally developing this I saw the same thing you did, and then tried with a larger image. I found that 512px filled the key. At the time I didn't probe this any further, but it would definitely be good to understand what's going on here.

These are just some of my thoughts off the top of my head. I'll take a look at the various problems and come back on this thread as soon as I can.

**Comment by pklaus** [comment]
Sounds great! I wish you a good evening!

**Comment by jamesridgway** [comment]
Summary of issues
- [ ] Create `.devdeck` if it does not exist
- [ ] Better validation of `output` for `devdeck.decks.volume_deck.VolumeDeck`
- [ ] Documentation for built-in controls
- [ ] Fix assets folder distribution
- [ ] Fix readme (CommandControl vs ClockControl)
- [ ] Investigate image size (72px vs 512px)

**Comment by jamesridgway** [comment]
Added a `MANIFEST.ini` and enabled `include_package_data` which should fix the assets issue for the next release

**Comment by jamesridgway** [comment]
Added better validation for `VolumeDeck` in `1.0.3`

**Comment by jamesridgway** [comment]
Built in controls are now documented here:
https://github.com/jamesridgway/devdeck/wiki/Controls

**Comment by jamesridgway** [comment]
Decks are also documented in the wiki and I have updated the README to link through to the relevant parts of the wiki

**Comment by jamesridgway** [comment]
@pklaus 

I have done some investigation on the images front, and for the moment I plan to leave this at 512px

I know the official elgato software states 72px, and I have investigated reducing the resolution to this but it does result in more pixelated graphics.

I think all of the issues you have listed should now be fixed under 1.0.5. 

If you have any further issues please open a new issue per problem.

Thanks!

**Comment by pklaus** [comment]
Awesome, thanks! Looking forward to testing this.
